internal/server/ip/tuntap.go

```diff
@@ -14,21 +16,35 @@ type Tuntap struct {
 
 // Add adds new tuntap interface.
 func (t *Tuntap) Add() error {
-	cmd := []string{"tuntap", "add", "name", t.Name, "mode", t.Mode}
+	var mode netlink.TuntapMode
+
+	switch t.Mode {
+	case "tun":
+		mode = netlink.TUNTAP_MODE_TUN
+	case "tap":
+		mode = netlink.TUNTAP_MODE_TAP
+	default:
+		return fmt.Errorf("invalid tuntap mode %q", t.Mode)
+	}
+
+	// TODO: there is TUNTAP_DEFAULTS and TUNTAP_MULTI_QUEUE_DEFAULTS in the netlink package, I don't know if these should be used
```

Comment:
Yeah, looking at the definitions and kernel doc, I think it'd make sense to set `flags` to `TUNTAP_DEFAULTS` and then if MultiQueue is requested to set it to `TUNTAP_MULTI_QUEUE_DEFAULTS` instead.